dvc/remote/base.py

```diff
@@ -847,9 +847,23 @@ def cache_exists(self, checksums, jobs=None, name=None):
         checksums = frozenset(checksums)
         prefix = "0" * self.TRAVERSE_PREFIX_LEN
         total_prefixes = pow(16, self.TRAVERSE_PREFIX_LEN)
-        remote_checksums = set(
-            map(self.path_to_checksum, self.list_cache_paths(prefix=prefix))
-        )
+        with Tqdm(
+            desc="Estimating size of "
+            + ("cache in '{}'".format(name) if name else "remote cache"),
+            bar_format=Tqdm.BAR_FMT_NOTOTAL,
+            unit="objects",
+        ) as pbar:
+            remote_checksums = set(
+                map(
+                    self.path_to_checksum,
+                    self.list_cache_paths(
+                        prefix=prefix,
+                        progress_callback=lambda n: pbar.update(
+                            n * total_prefixes
+                        ),
```

Comment:
If I use `unit_scale=total_prefixes` it scales the value of `n` properly after each `update()`, but this also causes the pbar to stop displaying the condensed metric prefixed values. Meaning the bar shows the total as `100000` rather than `1.0M` (even when I explicitly set `unit_divisor=1000`) - https://asciinema.org/a/XeDlDCqZoQyuDxgPsBvjtQQHV?t=12

Is there a way to make the units scale by `total_prefixes` (4096 in this example) but still keep the behavior to condense the displayed number by thousand/million/etc

Reply:
hmm no, as of now `unit_divisor` is ignored unless `unit_scale=True`.